lib/spack/spack/build_types.py

```diff
@@ -104,5 +106,6 @@ def get_build_type(spec):
     So we have a getter method that gets the class associated with the string
     """
     build_type = spec.variants.get('spack_build_type')
-    build_type = "null" if not build_type else build_type.value
-    return build_types[build_type](spec)
+    if build_type.value not in build_types:
+        tty.die("%s is not a valid build type" % build_type.value)
```

Comment:
We can set up the variant with a list of allowed values, so the Variant class will give a much clearer/earlier error message. Then we don't have to check the types here. I'll comment on that line for the change.